core/src/test/kotlin/ch/ergon/dope/resolvable/expression/CaseExpressionTest.kt

```diff
@@ -0,0 +1,290 @@
+        )
+
+        val actual = underTest.toDopeQuery()
+
+        assertEquals(expected, actual)
+    }
+
+    @Test
+    fun `should support searched case expression with multiple when then`() {
+        val expected = DopeQuery(
+            "CASE WHEN `first` THEN `stringField` WHEN `second` THEN `numberField` END",
+            emptyMap(),
+        )
+        val underTest = SearchedCaseExpression(
+            someBooleanField("first") to someStringField(),
+            mapOf(someBooleanField("second") to someNumberField()),
+        )
+
+        val actual = underTest.toDopeQuery()
+
+        assertEquals(expected, actual)
+    }
+
+    @Test
+    fun `should support searched else case expression with single when then`() {
+        val expected = DopeQuery(
+            "CASE WHEN `booleanField` THEN `stringField` ELSE `numberField` END",
+            emptyMap(),
+        )
+        val underTest = SearchedElseCaseExpression(
+            someBooleanField() to someStringField(),
+            elseCase = someNumberField(),
+        )
+
+        val actual = underTest.toDopeQuery()
+
+        assertEquals(expected, actual)
+    }
+
+    @Test
+    fun `should support searched else case expression with multiple when then`() {
+        val expected = DopeQuery(
+            "CASE WHEN `booleanField` THEN `stringField` WHEN `other2` THEN `numberField` ELSE `stringField` END",
+            emptyMap(),
+        )
+        val underTest = SearchedElseCaseExpression(
+            someBooleanField() to someStringField(),
+            mapOf(someBooleanField("other2") to someNumberField()),
+            someStringField(),
+        )
+
+        val actual = underTest.toDopeQuery()
+
+        assertEquals(expected, actual)
+    }
+
+    @Test
+    fun `should support aliased case expression with searched case`() {
+        val expected = DopeQuery(
+            "CASE WHEN `booleanField` THEN `stringField` AS `alias` END",
+            emptyMap(),
+        )
+        val underTest = AliasedCaseExpression(
+            alias = "alias",
+            unaliasedCaseExpression = SearchedCaseExpression(
+                someBooleanField() to someStringField(),
+            ),
+        )
+
+        val actual = underTest.toDopeQuery()
+
+        assertEquals(expected, actual)
+    }
+
+    @Test
+    fun `should support case function`() {
+        val numberField = someNumberField()
+        val expected = CaseClass(
+            numberField,
+        )
+
+        val actual = case(
+            numberField,
+        )
+
+        assertEquals(expected.toDopeQuery(), actual.toDopeQuery())
+    }
+
+    @Test
+    fun `should support when then function with simple case`() {
+        val numberField = someNumberField()
+        val case = someCaseClass(numberField)
+        val whenThen = someNumberField() to someStringField()
+        val expected = SimpleCaseExpression(numberField, whenThen)
+
+        val actual = case.`when`(whenThen.first, whenThen.second)
```

Comment:
Create the pair in the expected assignment (`SimpleCaseExpression(numberField, when to then`) so you do not have to use .first and .second

Reply:
done